server/util.go

```diff
@@ -224,23 +224,59 @@ func interceptConfigs(rootViper *viper.Viper) (*tmcfg.Config, error) {
 
 	conf.SetRoot(rootDir)
 
+	appCfgFilePath := filepath.Join(configPath, "app.toml")
+
+	appConf, err := config.ParseConfig(rootViper)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s: %w", appCfgFilePath, err)
+	}
+
+	if _, err := os.Stat(appCfgFilePath); os.IsNotExist(err) {
+		config.WriteConfigFile(appCfgFilePath, appConf)
+	}
+
+	// app.toml config file exist
+	rootViper.SetConfigType("toml")
+	rootViper.SetConfigName("app")
+	rootViper.AddConfigPath(configPath)
```

Comment:
This is already done once in the code, no need to repeat.

In my opinion, we should do the check inside `server/start.go`, in the startInProcess() function.

Reply:
An idea could be add a `ValidateBasic()` method on Config, so that in start.go, we check `ValidateBasic()`. ValidateBasic would contain the check that the min gas price is not empty.

Reply:
Thanks for your idea. I have implemented it.